src/include/mutex.h

```diff
@@ -22,6 +24,27 @@ pthread_mutex_lock(&NAME##_lock);
 
 #define CMARK_UNLOCK(NAME) pthread_mutex_unlock(&NAME##_lock);
 
+#else // building for windows
+
+#define _WIN32_WINNT 0x0600
+#include <synchapi.h>
+
+#define CMARK_DEFINE_ONCE(NAME) static INIT_ONCE NAME##_once = INIT_ONCE_STATIC_INIT;
+
+#define CMARK_RUN_ONCE(NAME, FUNC) \
+BOOL pending; BOOL status; \
+status = InitOnceBeginInitialize(&NAME##_once, 0, &pending, NULL); \
+if (status && pending) { FUNC(); } \
+InitOnceComplete(&NAME##_once, 0, NULL);
```

Comment:
The call to `InitOnceBeginInitialize` is not needed - you have the structure initialized during the definition on L32.  This should be:

```suggestion
#define CMARK_RUN_ONCE(NAME, FUNC) InitOnceExecuteOnce(&NAME##_once, FUNC, NULL, NULL)
```
The asynchronous initialization with `InitOnceBeginInitialize` is a bit more flexible and useful if you need to execute something without creating a new function.